dotCMS/src/main/java/com/liferay/portal/ejb/UserPersistence.java

```diff
@@ -231,31 +240,28 @@
 		}
 	}
 
+	@CloseDBIfOpened
 	protected com.liferay.portal.model.User findByPrimaryKey(String userId)
-		throws NoSuchUserException, SystemException {
+			throws NoSuchUserException, SystemException {
 		com.liferay.portal.model.User user = UserPool.get(userId);
-		Session session = null;
 
 		try {
 			if (user == null) {
-				session = openSession();
-
-				UserHBM userHBM = (UserHBM)session.load(UserHBM.class, userId);
-				user = UserHBMUtil.model(userHBM);
+				DotConnect dc = new DotConnect();
+				dc.setSQL(SEARCH_USER_BY_ID);
+				dc.addParam(userId);
+				List<Map<String, Object>> list = dc.loadObjectResults();
+				if(list.isEmpty()) {
+					throw new NoSuchUserException(userId.toString());

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') "userId" is already a string, there's no need to call "toString()" on it. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1858)